### PR TITLE
Send full commit SHA1 for each branch

### LINKED ISSUE #274: Send full commit SHA1 for each branch
As [suggested by @neelamgehlot](https://github.com/jupyterlab/jupyterlab-git/pull/270#discussion_r235574371), if we sent the SHA1 for each branch, then we wouldn't need to do parse the branch description as done in #270

### PR DESCRIPTION
**Pull request by neelamgehlot**
Fixes #274

### CODE REVIEW

File: tests/test-components/PastCommitNode.spec.tsx [hunk 1]
```diff
@@ -29,13 +31,15 @@ describe("PastCommitNode", () => {
       is_remote_branch: false,
       name: "name3",
       upstream: "upstream",
+      top_commit: "2414721hijklmnopqrstuvwxyz01234567890123",
```
Comment (saulshanabrook) [inline tests/test-components/PastCommitNode.spec.tsx]: This should be changed to `2414721b194453f058079d897d13c4e377f92dc6` so it matches.

File: tests/test-components/PastCommitNode.spec.tsx [hunk 2]
```diff
@@ -29,13 +31,15 @@ describe("PastCommitNode", () => {
...
       tag: "v1.0.4-14-g2414721"
     },
     {
       is_current_branch: false,
       is_remote_branch: true,
       name: "name4",
       upstream: "upstream",
+      top_commit: "2414721hijklmnopqrstuvwxyz01234567890123",
```
Comment (saulshanabrook) [inline tests/test-components/PastCommitNode.spec.tsx]: Same here `2414721b194453f058079d897d13c4e377f92dc6`

File: tests/test-components/PastCommitNode.spec.tsx (hunk 1 above)
Comment (neelamgehlot) [reply tests/test-components/PastCommitNode.spec.tsx]: Done

File: tests/test-components/PastCommitNode.spec.tsx (hunk 2 above)
Comment (neelamgehlot) [reply tests/test-components/PastCommitNode.spec.tsx]: Done

File: tests/test-components/PastCommitNode.spec.tsx [hunk 3]
```diff
@@ -29,13 +31,15 @@ describe("PastCommitNode", () => {
...
       upstream: "upstream",
+      top_commit: "2414721b194453f058079d897d13c4e377f92dc6",
       tag: "v1.0.4-14-g2414721"
     },
     {
       is_current_branch: false,
       is_remote_branch: true,
       name: "name4",
       upstream: "upstream",
+      top_commit: "2414721b194453f058079d897d13c4e377f92dc6",
```
Comment (jaipreet-s) [inline tests/test-components/PastCommitNode.spec.tsx]: WDYT about adding some assertions on this new `prop` so that its intended side-effect is clear?

Comment (neelamgehlot) [reply tests/test-components/PastCommitNode.spec.tsx]: Talked offline with @jaipreet-s , all the test cases are already using the new prop, so there is no need to add any more assertions.

**Pull request merged**